### ISSUE TITLE
Battery drain 

### ISSUE THREAD

**Issue by GianoMayne** [koreader/koreader#2854]
* KOReader version: Nightly build v2015.11-1023-g773e24f
* Device: Kobo Aura H2O

With the latest nightly build, I've noticed a problem: my Kobo was fully charged when I switched it to sleep mode. When I powered it on the next day, I noticed that the battery has dropped to around 40%.

I can provide any log file, just tell me where I can locate on my Kobo e-reader.

**Comment by AlanSP1** [comment]
Log file is in koreader root (depends where you installed it.

It looks like this is problem with charging device, more info you can found here: https://github.com/koreader/koreader/issues/2239

It is a long read, but also very informative about problems with sleep. I think that charging still isn't sorted out, but many things are. Workaround is that after charging you unplug device and turn it on and then off (put to sleep, not shut off).

**Comment by oldhasbeen** [comment]
Try to observe this for a longer period. It would be a newly introduced issue if it was due to false sleep.
It is true that the device will wake up when there is a USB event (like plug in our out of a charger but it would go back to true sleep after something like 17 seconds if left undisturbed. My H2O has not shown such behaviour since a very long time now. I run also currently the 1023 and have checked the known false sleep triggers after you posted your observation. Have however not come across any false sleep.

**Comment by GianoMayne** [comment]
Thanks for the indications. 
For the moment I haven't yet identified the cause. The battery issue has started when I've updated the nightly build around two weeks ago. Before, I didn't have any problems (and I don't remember unfortunately the build number).

I usually launch Nickel to order to allow e-book synchronisation with Calibre and battery charging.
I've a sleep cover which causes lots of issues with automatic power-on and sleep in Nickel mode so I had to disable this feature both in Nickel and Koreader configurations. 

I've tried to use the system statistics of Koreader, and I've found an interesting fact: the sleeps counter, in a certain moment, was growing dramatically. I haven't yet found the precise reason. Also the "up hours" is somewhat exaggerated as I've read around 2h today.

> KOReader started at Fri May 5 06:28:14 2017 
Last suspend time	Fri May 5 17:37:45 2017
Last resume time	Fri May 5 17:37:46 2017
Up hours	11.16
Counters
wake-ups	16
**sleeps	20494**
charge cycles	0
discharge cycles	0
Process
ID	804
Processor usage %	3.62
Threads	1
Virtual memory (MB)	111.36
RAM usage (MB)	30.45

Will continue my investigations. For the moment the log file wasn't helpful yet.

**Comment by a-kohout** [comment]
The problem on Kobos with the sleepcover is that it is controlled in software only.
Even if disabled, if the magnet moves, the system wakes up (without anything shown on the display), sees it shouldn't do anything and goes back to sleep.
With my 1st gen Kobo Glo HD sleepcover which has lots of play, leaving the reader in a backpack and walking around all day often manages to completely drain the battery. Instead of having to turn the device off I attached the cover upside down so it opens like a hebrew language book. Obviously cutting out the magnet would work too.

**Comment by AlanSP1** [comment]
@GianoMayne if you check issue that @oldhasbeen linked, you'll see he noticed that autosleep seems broken for some reason in latest builds. Older versions worked fine. So, it looks that something needs to be done again to solve this problem.

**Comment by slow-mo** [comment]
BTW it seems to me that since the last two nightly builds koreader is draining a lot of battery on my kindle PW2. I charged it yesterday in the morning and now it's at 90%. I'm not sure yet if it happens in sleep mode or not, but I'd say at the moment that the issue is not limited to sleep mode.

**Comment by oldhasbeen** [comment]
Please note that I have posted an update on my autosleep problem. It is perhaps a red herring.

**Comment by GianoMayne** [comment]
I can definitely confirm that the sleep cover is the root cause, especially the magnet piece which retains the standing holder (which already causes problems with autosleep with Nickel).

I've removed the sleep cover and replaced it with a bubble envelope and now I don't have any problems. I'll consider to replace the magnet piece for a velcro.

As @AlanSP1 said, I have also the same impression that the two last builds break the auto sleep system, as I didn't have any issues before.

**Comment by Frenzie** [comment]
@GianoMayne That seems unlikely as the only thing even vaguely sleep-related in the past two weeks or so was fa12f488e8c47340440cd4edcd5f6a4cff681844.

**Comment by GianoMayne** [comment]
@Frenzie Sorry, I should have indicated that I don't update regularly the nightly build of KOReader (perhaps once a month). I don't really remember the build number where I didn't have issues with the sleep cover.

**Comment by Frenzie** [comment]
That would've been one to two months ago. Hard to say if it's related, especially as everything's working fine on my H2O. It even auto-slept a week or two ago when I forgot to suspend it manually.

@Hzj-jie Any thoughts?

**Comment by KenMaltby** [comment]
There is a way to have KOReader ignore the magnetic sleep cover:
After KOReader has been started and the settings files created:

Killing the magnetic cover?
you can set  ["ignore_power_sleepcover"] = true  in  settings.reader.lua  

Luck;
Ken

**Comment by GianoMayne** [comment]
@KenMaltby Thank you very much, but it's already set in my settings file... I remember to have set it to cope with my sleep cover.

**Comment by Hzj-jie** [comment]
It's a known issue that Kobo may randomly wake up sometimes. So the number of sleeps would be larger than number of wake-ups as we directly put the device into sleep again without involving wake-up event. Say on my device, it's 14 vs 34. But if the # of sleeps went too high as 20k, something else may be wrong.
p.s. "ignore_power_sleepcover" won't stop the magnetic cover from waking up the device, KOReader uses this setting to ignore the events. i.e. putting the device back into sleep.

**Comment by KenMaltby** [comment]
So that setting might be counter productive?

**Comment by Hzj-jie** [comment]
It's not counter productive, but for a different purpose. The setting gives user a chance to disable the feature. Otherwise the device may accidentally wake up when moving around a phone or a kindle, e.g. any devices with a magnet.

**Comment by GianoMayne** [comment]
Finally found the root cause of battery drain : when setting my Kobo to sleep mode, the front light remains enabled. The light is very hardly noticeable on day, it's why I didn't noticed it before yesterday evening.

I've also noticed that even if the front light is completely disabled, the light is anyway on. The minimal value can not be lower than 1.

Is there a reason that the value can't be set to 0 (so it would be completely off) ?

**Comment by Frenzie** [comment]
It used to be able to, but now the brightness and on/off are separate. However, it really should turn off when suspending regardless. That's a definite bug, but one that doesn't occur on my H2O as far as I can tell.

**Comment by a-kohout** [comment]
I just noticed the same last night on my Glo HD. This is a regression, the backlight used to turn off properly in the past. I'll try to find the build that introduced this tonight.

**Comment by Frenzie** [comment]
I'd suspect #2941. Atm I'm using v2016.11-1036-g9e346f5 which predates that.

**Comment by GianoMayne** [comment]
I've rolled back koreader to v2015.11-894-gd168db9.
I can confirm that this version allows front light value to be set to 0.

**Comment by a-kohout** [comment]
I've created a new issue #2973. The last correctly working version is `v2015.11-1057-ged0ba67`

**Comment by oldhasbeen** [comment]
version 1071 Aura H2O:
I can confirm frontlight issue: cannot swipe light easily to 1% but only 2%. Going lower turns frontlight off.  However swiping very careful (is delicate but doable without turning light off)) lower than 2% to get 1% will show L:Off but light is then in fact on at what looks like 1% level. Toggling left lower button gives then message "Frontlight is off" at every touch without changing the frontlight (always 1%). To enable toggle I need to swipe back up to at least 2%.
Suspending turns light off when in 1% non-toggable state and after waking up light is off and toggle works at the 2% level then..

**Comment by GianoMayne** [comment]
My Aura H2O was completely drained today once again, and the build is 1077.
The big problem is that it's very very hard to notice if the front light is really shut down or not, unless I'll go to a darker room (which isn't always an option). 

What is also bothering me is that settings (for example font size and margins) are lost again.

**Comment by cramoisi** [comment]
got this this morning : 95 % yest - 15 this morning. I got this before but the option given by ken seems to works things out. I only forgot to redo it after my reinstallation.

**Comment by Frenzie** [comment]
@GianoMayne Is this still an issue after #3809?

**Comment by Frenzie** [comment]
Closing due to inactivity. Feel free to reopen with more info.
